components/x-follow-button/src/FollowButton.jsx

```diff
@@ -0,0 +1,74 @@
+import { h } from '@financial-times/x-engine';
+import { withActions } from '@financial-times/x-interaction';
+
+import Input from './Input';
+import Button from './Button';
+
+import styles from './styles/main.scss';
+
+const followButtonActions = withActions((props) => ({
+	onSubmitAction(event) {
+		event.preventDefault();
+		console.log('the follow button was pressed');
+		console.log('the default event was prevented');
+		console.log(`the action suppose to trigger the submission of the form with "${event.target.method}" method and "${event.target.action}" action`);
+	},
+	onClickAction(event) {
+		event.preventDefault();
+		return ({isSelected}) => ({
+			isSelected: !isSelected
+		});
+	}
+}));
+
+const getFormAction = (conceptId, followPlusDigestEmail, isSelected) => {
+	if (followPlusDigestEmail) {
+		return `/__myft/api/core/follow-plus-digest-email/${conceptId}?method=put`
+	} else if (isSelected) {
+		return `/__myft/api/core/followed/concept/${conceptId}?method=delete`
+	} else {
+		return `/__myft/api/core/followed/concept/${conceptId}?method=put`
+	}
+};
+
+const BaseButton = ({
+	conceptId,
+	csrfToken,
+	followPlusDigestEmail,
+	isSelected,
+	extraButtonClasses,
+	variant,
+	alternateText,
+	buttonText,
+	actions,
+	name
+}) => (
+	<form
+		className="n-myft-ui n-myft-ui--follow {{extraClasses}}"
```

Comment:
are these classes needed? i don't see them in the x-follow-button styles. if they're something for n-myft-ui or something else external, perhaps you could add support for `className` to be passed in?